backend/eventloop/db_reconciler.go

```diff
@@ -444,6 +469,360 @@ func cleanRowFromDB(ctx context.Context, dbQueries db.DatabaseQueries, id string
+				offSet, offSet+appRowBatchSize))
+			break
+		}
+
+		// Break the loop if no entries are left in table to be processed.
+		if len(listOfApplicationsFromDB) == 0 {
+			log.Info("All Application entries are processed by cleanOrphanedEntriesfromTable_Application.")
+			break
+		}
+
+		// Iterate over batch received above.
+		for _, appDB := range listOfApplicationsFromDB {
+
+			// Check if application has entry in DTAM table, if not then delete the application
+			// If created time is less than waitTimeforRowDelete then ignore dont delete even if ACTDM entry is missing.
+			if !slices.Contains(listOfAppsIdsInDTAM[string(dbType_Application)], appDB.Application_id) &&
+				time.Since(appDB.Created_on).Hours() > waitTimeforRowDelete {
+
+				// Fetch Application from DB and convert Spec into an Object as we need namespace for creating Operations CR.
+				var appArgo fauxargocd.FauxApplication
+				isApplicationPresent := true
+				if err := yaml.Unmarshal([]byte(appDB.Spec_field), &appArgo); err != nil {
+					log.Error(err, "Error occurred in cleanOrphanedEntriesfromTable_Application while unmarshalling application: "+appDB.Application_id)
+					isApplicationPresent = false
+				}
+
+				_, err := dbQueries.DeleteApplicationById(ctx, appDB.Application_id)
+				if err != nil {
+					log.Error(err, "Error occurred in cleanOrphanedEntriesfromTable_Application while cleaning Application entry "+appDB.Application_id+" from DB.")
+					continue
+				} else {
+					log.Info("Application entry: " + appDB.Application_id + " is successfully deleted by cleanOrphanedEntriesfromTable_Application.")
+				}
+
+				// Skip if Application was not found, as we need namespace for creating Operation CR.
+				if isApplicationPresent {
+					// Create k8s Operation to delete related CRs using Cluster Agent
+					createOperation(ctx, appDB.Engine_instance_inst_id, appDB.Application_id, appArgo.Namespace, db.OperationResourceType_Application, dbQueries, client, log)
+				}
+			}
+		}
+
+		// Skip processed entries in next iteration
+		offSet += appRowBatchSize
+	}
+}
+
+// getListOfCRIdsFromTable loops through DTAMs in database and returns list of resource IDs for each CR type (i.e. RepositoryCredential, ManagedEnvironment, SyncOperation) .
+func getListOfCRIdsFromTable(ctx context.Context, dbQueries db.DatabaseQueries, tableType string, log logr.Logger) map[string][]string {
+
+	offSet := 0
+
+	// Create Map to store resource IDs according to type, Ex: {"RepositoryCredential" : [], "ManagedEnvironment" : [], "SyncOperation" : []}
+	crIdMap := make(map[string][]string)
+
+	// Continuously iterate and fetch batches until all entries of table processed.
+	for {
+		if offSet != 0 {
+			time.Sleep(sleepIntervalsOfBatches)
+		}
+
+		// If resource type is Application then get list of Application IDs from DTAM table.
+		if tableType == "Application" {
```

Comment:
Can we use the `dbTableName` type?
```suggestion
		if tableType == dbType_Application {
```

Reply:
Yes, initially I used this only, but we don't have any `dbTableName` for the else block, which will execute for `RepositoryCredential`, `ManagedEnvironment`, `SyncOperation` tables. I wanted to use a better name but couldn't come up with one, thought about `dbType_Multiple dbTableName = "Multiple"` but didn't look good to me.
https://github.com/redhat-appstudio/managed-gitops/blob/b04b7dac2c1c5545f4ca9d1ff7578576a2dc20ae/backend/eventloop/db_reconciler.go#L787

Reply:
Actually there are few more clean jobs I have to implement, so I was thinking of using same function and in that case there would be more `if..else` blocks will be added, but lets change it when new logic is added for now I removed 2nd if condition.